Source/DafnyServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -458,34 +513,65 @@ public class DafnyModel {
           utfCode = ((Model.Integer)fCharToInt.OptEval(elt)).AsInt();
           return "'" + char.ConvertFromUtf32(utfCode) + "'";
         }
-        return "?#" + GetShortElementId(elt);
+        return GetUnspecifiedValue(elt, "char");
       }
       if (fType.OptEval(elt) == fReal.GetConstant()) {
         if (fU2Real.OptEval(elt) != null) {
           return CanonicalName(fU2Real.OptEval(elt));
         }
-        return "?#" + GetShortElementId(elt);
+        return GetUnspecifiedValue(elt, "real");
       }
       if (fType.OptEval(elt) == fBool.GetConstant()) {
         if (fU2Bool.OptEval(elt) != null) {
           return CanonicalName(fU2Bool.OptEval(elt));
         }
-        return "?#" + GetShortElementId(elt);
+        return GetUnspecifiedValue(elt, "bool");
       }
       if (fType.OptEval(elt) == fInt.GetConstant()) {
         if (fU2Int.OptEval(elt) != null) {
           return CanonicalName(fU2Int.OptEval(elt));
         }
-        return "?#" + GetShortElementId(elt);
+        return GetUnspecifiedValue(elt, "int");
       }
       return "";
     }
 
-    private int GetShortElementId(Model.Element element) {
-      if (!shortElementIds.ContainsKey(element.Id)) {
-        shortElementIds[element.Id] = shortElementIds.Count;
+    private string GetUnspecifiedValue(Model.Element element, string primitiveType) {
+      if (reservedValuesMap.ContainsKey(element)) {
+        return reservedValuesMap[element];
+      }
+
+      if (primitiveType == "bool") {
+        if (!reservedValues["bool"].Contains("true")) {
+          reservedValues["bool"].Add("true");
+          reservedValuesMap[element] = "true";
+        } else {
+          reservedValuesMap[element] = "false";
+        }
+        return reservedValuesMap[element];
+      }
+
+      var i = 0;
+      if (primitiveType == "char") {
+        i = 33; // 33 is the first non special character (excluding space)
```

Comment:
I wonder if we should maybe be a bit more user friendly, and first try to inject some kind of more reasonable characters, such as letters. Not a big deal, just something to maybe think about.

Reply:
Now starting from character `'A'` - will fix this in the paper along with the badges once we get the instructions on how to add them.